app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -34,31 +33,56 @@ def logout
 
     private
 
+    def saml_settings
+      @_saml_settings ||= OneLogin::RubySaml::Settings.new(Saml::Config.new.settings.dup)
+    end
+
     def auth_hash
       request.env['omniauth.auth']
     end
 
+    def validate_slo_response
+      slo_response = OneLogin::RubySaml::Logoutresponse.new(params[:SAMLResponse], saml_settings)
+      if slo_response.validate
+        flash[:notice] = t('omniauth.logout_ok')
+        redirect_to root_url
+      else
+        flash[:alert] = t('omniauth.logout_fail')
+        render :failure
+      end
```

Comment:
Looks like this is not tested. By the way, I can tell because I'm using Code Climate's awesome browser extension, which you have to sign up to get, but I got access pretty quickly. https://codeclimate.com/browser-extension